plugins/binaryninja/mui/hook_manager.py

```diff
@@ -180,3 +181,19 @@ def save_hooks(self) -> None:
             view=bv,
             scope=SettingsScope.SettingsResourceScope,
         )
+
+
+@dataclass(frozen=True)
+class CustomHookIdentity:
```

Comment:
I feel like there could be a more appropriate name for this but don't have any concrete ideas myself 🤔

Reply:
I think this is fine